src/ML/include/BipedalLocomotion/ML/velMANNTrajectoryGenerator.h

```diff
@@ -0,0 +1,184 @@
+/**
+ * @file velMANNTrajectoryGenerator.h
+ * @authors Paolo Maria Viceconte, Giulio Romualdi
+ * @copyright 2023 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_ML_VEL_MANN_TRAJECTORY_GENERATOR_H
+#define BIPEDAL_LOCOMOTION_ML_VEL_MANN_TRAJECTORY_GENERATOR_H
+
+#include <memory>
+
+#include <Eigen/Dense>
+
+#include <BipedalLocomotion/Contacts/ContactPhaseList.h>
+#include <BipedalLocomotion/ML/velMANNAutoregressive.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+namespace BipedalLocomotion
+{
+namespace ML
+{
+
+/**
+ * Input of the planner.
+ */
+struct velMANNTrajectoryGeneratorInput : public velMANNAutoregressiveInput
+{
+    /** Index to the merge point considered to attach the new trajectory */
+    std::size_t mergePointIndex;
+};
+
+/**
+ * Output of the trajectory planner
+ */
+struct velMANNTrajectoryGeneratorOutput
+{
+    /** CoM trajectory expressed in the inertial frame */
+    std::vector<Eigen::Vector3d> comTrajectory;
+
+    /** CoM velocity trajectory expressed in the inertial frame */
+    std::vector<Eigen::Vector3d> comVelocityTrajectory;
+
+    /** base velocity trajectory expressed in mixed representation */
+    std::vector<manif::SE3d::Tangent> baseVelocityTrajectory;
+
+    /** Centroidal angular momentum trajectory expressed in the mixed representation. */
+    std::vector<Eigen::Vector3d> angularMomentumTrajectory;
+
+    std::vector<Eigen::VectorXd> jointPositions; /**< Joints position in radians */
+    std::vector<Eigen::VectorXd> jointVelocities; /**< Joint velocities in radians */
+    std::vector<manif::SE3d> basePoses; /**< Vector containing the base pose for each instant. */
+
+    std::vector<std::chrono::nanoseconds> timestamps; /**< Vector containing the time stamps. */
+
+    Contacts::ContactPhaseList phaseList; /**< List of the contact phases */
+};
+
+/**
+ * velMANNTrajectoryGenerator is a class that uses velMANNAutoregressive to generate a kinematically
+ * feasible trajectory for humanoid robots. The planner will generate a trajectory which duration is
+ * equal to `slow_down_factor * time_horizon` seconds.
+ * @subsection mann_trajectory_generator velMANN trajectory generator
+ * The diagram illustrates the utilization of the velMANNAutoregressive within the
+ * velMANNTrajectoryGenerator class.
+ * <img src="https://user-images.githubusercontent.com/16744101/239922103-1f23e08b-0091-475a-8879-61af66399c62.png" alt="mann_trajectory_generator">
```

Comment:
This image could be confusing since it refers to a different implementation of MANN I think you can remove it and write that the calling way is similar to `ML::MANNTrajectoryGenerator`

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/commit/8bf5fe2491861d253514364478ce5f8829db23c7.